src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php

```diff
@@ -101,17 +101,17 @@ public static function typesProvider()
             [
                 'files',
                 [
-                    new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_OBJECT, false, 'SplFileInfo')),
+                    new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)], new Type(Type::BUILTIN_TYPE_OBJECT, false, 'SplFileInfo')),
                     new Type(Type::BUILTIN_TYPE_RESOURCE),
                 ],
                 null,
                 null,
             ],
             ['bal', [new Type(Type::BUILTIN_TYPE_OBJECT, false, 'DateTimeImmutable')], null, null],
             ['parent', [new Type(Type::BUILTIN_TYPE_OBJECT, false, 'Symfony\Component\PropertyInfo\Tests\Fixtures\ParentDummy')], null, null],
-            ['collection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_OBJECT, false, 'DateTimeImmutable'))], null, null],
-            ['nestedCollection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING, false)))], null, null],
-            ['mixedCollection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, null, null)], null, null],
+            ['collection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)], new Type(Type::BUILTIN_TYPE_OBJECT, false, 'DateTimeImmutable'))], null, null],
+            ['nestedCollection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)], new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)], new Type(Type::BUILTIN_TYPE_STRING, false)))], null, null],
+            ['mixedCollection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)], null)], null, null],
```

Comment:
@mtarld talked about the failures during the hackathon. What we would have preferred as the solution is changing only this line to:

```suggestion
            ['mixedCollection', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), null)], null, null],
```

That would mean not having to do any changes to prepare for the next `phpdocumentor/type-resolver` release and would also mean to be consistent with the other collection examples.

However, I found no way to distinguish `mixed[]` and `array` on released versions of that package meaning that we would have to bump constraint to have consistently passing tests.

Now, we can get the CI green by enforcing that the key type for arrays is `int|string` which is technically correct but may break existing code that relies on the key being `null`. I am not sure how we want to proceed here (if we are fine with the current solution, we should probably align the other extractors).

Reply:
looks like I found a way to restore the previous (even if not fully correct) behaviour, see #62577 as an alternative (now checking if we can achieve the same for 7.3+ when we use the TypeInfo component under the hood)